src/components/profile/images/DeleteImage.js

```diff
@@ -1,36 +1,62 @@
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
 import PropTypes from 'prop-types';
 
 import Dialog from '@material-ui/core/Dialog';
 import DialogTitle from '@material-ui/core/DialogTitle';
 import DialogActions from '@material-ui/core/DialogActions';
+import CircularProgress from '@material-ui/core/CircularProgress';
 import Button from '@material-ui/core/Button';
-import Typography from '@material-ui/core/Typography';
 import { withStyles } from '@material-ui/core';
+import { deleteImage, resetImageProps } from '../../../actions/images';
+import { isLoading } from '../../../reducers/display';
+import types from '../../../utils/types';
 
 const styles = theme => ({
   errorStyle: {
     margin: theme.spacing.unit,
   },
+  loadingStyle: {
+    marginRight: '20px',
+  },
 });
 
 class DeleteImage extends Component {
+  constructor(props) {
+    super(props);
+    this.handleDelete = this.handleDelete.bind(this);
+  }
+  handleDelete = () => {
+    this.props.deleteImage(this.props.image);
+  };
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.deleteSuccess !== this.props.deleteSuccess && this.props.deleteSuccess) {
+      // this is called many times in a row because the update is not fast enough
```

Comment:
Is this comment for the if statement or the `handleCloseDelete`?

Reply:
It's that the if statement should only be evaluate to true once since we change the props in the body of the if statement. However, the body of the if statement is called many times.

Reply:
I made this another issue #36 since it does not impede on the user experience for now.